tservice/src/seal.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2022 The MobileCoin Foundation
+//! Functions used for sealing and unsealing of secrets
+
+use alloc::{vec, vec::Vec};
+pub use core::ptr;
+pub use mc_sgx_core_types::{Error, Result};
+pub use mc_sgx_tservice_sys::sgx_calc_sealed_data_size;
+pub use mc_sgx_tservice_sys_types::sgx_sealed_data_t;
+pub use mc_sgx_tservice_types::Sealed;
+use mc_sgx_util::ResultInto;
+
+/// Sealed data builder
+#[derive(Debug, Clone, Hash, PartialEq, Eq, Default)]
+pub struct SealedBuilder<T> {
+    /// The data to be encrypted/sealed
+    data: T,
+
+    /// The MAC text that will not be encrypted
+    mac: Option<T>,
```

Comment:
Aside: this should either be called `aad` or `mac_text`, and the comment should really indicate that they mean the same thing.

Reply:
renamed to aad